src/gui/render.cpp

```diff
@@ -896,6 +1283,25 @@ void RENDER_AddConfigSection(const config_ptr_t& conf)
 	Section_prop* sec = conf->AddSection_prop("render",
 	                                          &RENDER_Init,
 	                                          changeable_at_runtime);
+
+	MAPPER_AddHandler(toggle_stretch_axis,
+	                  SDL_SCANCODE_UNKNOWN,
+	                  0,
+	                  "stretchax",
+	                  "Stretch Axis");
+
+	MAPPER_AddHandler(increase_viewport_stretch,
+	                  SDL_SCANCODE_UNKNOWN,
+	                  0,
+	                  "incstretch",
+	                  "Inc Stretch");
+
+	MAPPER_AddHandler(decrease_viewport_stretch,
+	                  SDL_SCANCODE_UNKNOWN,
+	                  0,
+	                  "decstretch",
+	                  "Dec Stretch");
+
```

Comment:
Suggest a single hotkey that activates the stretch controls.
The controls themselves can be fixed to something like:
 - left & right arrow keys: horizontal shrink and stretch.
 - down & up arrow keys: vertical shrink and stretch.

(this is how the joystick shift and scale controls work using a single "throw switch" that gives you the control, after which you user the arrow keys to very intuitively tweak the joystick's shift and scale wheels along all four axis)

The single activator button is just like flipping open the CRT control panel:

![2023-11-30_08-01](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/0c2952bf-01bb-4c16-bd2f-47c39f643037)

Once the panel is flipped open, you've got your left-right and up-down buttons for very intuitive realtime tweaking:

![2023-11-30_08-10](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/9869a641-0f05-40cc-b174-3d1c0e4f45fb)

Then the user hits Ctrl+Alt+S again to disable the stretch settings.

And the arrow keys go back to normal.

The user would be able to change the Ctrl+Alt+S to something else, just in case this is an actual in-game key. But I think Ctrl+Alt+S is rarely used, so it's probably reasonable to have this available by default to access the stretch settings.

Reply:
That's a nifty idea, but the problem is I don't think you can do that at runtime. Meaning once you have added a key handler with `MAPPER_AddHandler`, you can't remove it again. I really don't wanna make mapper changes as part of this PR! 😅 (or touch the mapper at all except for basic stuff, as a matter of fact).

You kinda worked around this with the `use_joy_calibration_hotkeys` setting. I guess the idea is you add that to your config, calibrate the joystick, then remove it from your config once you're arrived to the correct settings. A bit clunky, but does the job 😄 

Also, because all those cursor + modifier combos need to be defined as separate hotkeys, there's a large-scale "hotkey pollution" going on with `use_joy_calibration_hotkeys` enabled:

<img width="1215" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/3fd58cdc-59ca-42e6-b328-b849de4d45c0">

So... dunno. I'm kinda inclined to leave it as is because I don't see the joy approach to be really superior. Plus the current approach is consistent with the CGA composite colour adjustments which works similarly to how I've done it.

Overall, It's one of those things that we'll be able to do a lot better with the new mapper. This does the job for now, and we can always iterate on it and refine it in the future.

Reply:
To be clear @kcgen, I like your idea a lot, and this would be the ultimate goal for keyboard-driven viewport adjustments, no disagreements. We could also throw the Shift modifier into the mix to allow 5% increments instead of just 1%, and ideally, we'd also have keyrepeat, just like on a real monitor that has digital width/height controls. Now you must press the adjustment keys many many times, but that's how the mapper works...

But we need some sane infrastructure to go in first, yeah 😅